javaslang/src/main/java/javaslang/collection/IndexedSeq.java

```diff
@@ -220,19 +221,30 @@ default int lastIndexWhere(Predicate<? super T> predicate, int end) {
 
     @Override
     default Iterator<T> reverseIterator() {
-        return new AbstractIterator<T>() {
-            private int i = IndexedSeq.this.length();
+        return new AbstractIterator<T>() { /*@formatter:off*/
+            final ListIterator<T> delegate = IndexedSeq.this.listIterator(IndexedSeq.this.length());
+            @Override public boolean hasNext() { return delegate.hasPrevious(); }
+            @Override public T getNext()       { return delegate.previous(); }
```

Comment:
> Don't be too obsessed with formatting. We have a formatter, so just apply it and forget about formatting.

Removed custom formatting in a separate commit.

> The API should be expressive and concise on its own, with 'default' formatting.

I put it in a separate commit so that you can decide which is more "expressive and concise".